src/EditorFeatures/Core/InheritanceMargin/InheritanceMarginDefinitionItemExtensions.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.FindUsages;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.Editor.InheritanceMargin
+{
+    internal static class InheritanceMarginDefinitionItemExtensions
+    {
+        /// <summary>
+        /// Create the DefinitionItem based on the numbers of locations for <paramref name="symbol"/>.
+        /// If there is only one location, create the DefinitionItem contains only the documentSpan or symbolKey to save memory.
+        /// Because in such case, when clicking InheritanceMarginGlpph, it will directly navigate to the symbol.
```

Comment:
i genuinely don't really like this idea.  i often use the glhyph to understand the hierarchy, not necessarily to navigate automatically.

Reply:
oh.  i think i misunderstood the comment hre.  you still show all the items when hovering over the margin item.  but clicking on it just navigates straight to the item, it doesn't need to show it in the FAR window.

Reply:
Yes